staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -690,44 +714,69 @@ func (s *Controller) nodeSyncLoop() {
 	// round.
 	s.servicesToUpdate = s.cache.allServices()
 	numServices := len(s.servicesToUpdate)
-	s.servicesToUpdate = s.updateLoadBalancerHosts(s.servicesToUpdate, newHosts)
+	s.servicesToUpdate = s.updateLoadBalancerHosts(s.servicesToUpdate, workers)
 	klog.V(2).Infof("Successfully updated %d out of %d load balancers to direct traffic to the updated set of nodes",
 		numServices-len(s.servicesToUpdate), numServices)
 
 	s.knownHosts = newHosts
 }
 
+// nodeSyncService syncs the nodes for one load balancer type service
+func (s *Controller) nodeSyncService(svc *v1.Service) bool {
+	if svc == nil || !wantsLoadBalancer(svc) {
+		return false
+	}
+	klog.V(4).Infof("nodeSyncService started for service %s/%s", svc.Namespace, svc.Name)
+	hosts, err := listWithPredicate(s.nodeLister, s.getNodeConditionPredicate())
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("failed to retrieve node list: %v", err))
+		return true
+	}
+
+	if err := s.lockedUpdateLoadBalancerHosts(svc, hosts); err != nil {
+		runtime.HandleError(fmt.Errorf("failed to update load balancer hosts for service %s/%s: %v", svc.Namespace, svc.Name, err))
+		return true
+	}
+	klog.V(4).Infof("nodeSyncService finished successfully for service %s/%s", svc.Namespace, svc.Name)
+	return false
+}
+
 // updateLoadBalancerHosts updates all existing load balancers so that
-// they will match the list of hosts provided.
+// they will match the latest list of nodes with input number of workers.
 // Returns the list of services that couldn't be updated.
-func (s *Controller) updateLoadBalancerHosts(services []*v1.Service, hosts []*v1.Node) (servicesToRetry []*v1.Service) {
-	for _, service := range services {
-		func() {
-			if service == nil {
-				return
-			}
-			if err := s.lockedUpdateLoadBalancerHosts(service, hosts); err != nil {
-				runtime.HandleError(fmt.Errorf("failed to update load balancer hosts for service %s/%s: %v", service.Namespace, service.Name, err))
-				servicesToRetry = append(servicesToRetry, service)
-			}
-		}()
+func (s *Controller) updateLoadBalancerHosts(services []*v1.Service, workers int) (servicesToRetry []*v1.Service) {
+	if len(services) == 0 {
```

Comment:
You don't need this if - this is handled in ParallelizeUntil automatically

Reply:
I want this to stop log spamming though. But I guess. V(4) is high enough that in production the logs would not show up.

Reply:
At V(4) we already have a gazillions log. Let's remove it.